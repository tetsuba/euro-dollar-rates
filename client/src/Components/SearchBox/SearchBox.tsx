import React from 'react'
import {InputElementType} from "../../utils/types"

interface PropTypes {
    label: string
    inputFunc: (e: InputElementType) => void
    text: string
}

export default function SearchBox(props: PropTypes): JSX.Element {
    const { label, inputFunc, text } = props
    return (
        <div className="input-group input-group-lg">
            <span className="input-group-text" id="inputGroup-sizing-lg">
                {label}
            </span>
            <input
                data-testid="search-input"
                value={text}
                onChange={inputFunc}
                type="text"
                className="form-control"
                aria-label="Sizing example input"
                aria-describedby="inputGroup-sizing-lg"
            />
        </div>
    )
}
