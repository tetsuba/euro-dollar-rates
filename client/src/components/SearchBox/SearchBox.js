import React from 'react'

export default function SearchBox(props) {
    const { label, inputFunc, text } = props
    return (
        <div className="input-group input-group-lg">
            <span className="input-group-text" id="inputGroup-sizing-lg">
                {label}
            </span>
            <input
                value={text}
                onChange={inputFunc}
                type="text"
                className="form-control"
                aria-label="Sizing example input"
                aria-describedby="inputGroup-sizing-lg"
            />
        </div>
    )
}
