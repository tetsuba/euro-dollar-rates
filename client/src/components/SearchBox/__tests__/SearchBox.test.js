import {render} from '@testing-library/react'
import SearchBox from '../SearchBox'

describe('SearchBox', () => {
  const props = {
    label: 'Search Box',
    text: '',
    inputFunc: () => {}
  }

  test('rendered', () => {
    const { asFragment } = render(<SearchBox {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})