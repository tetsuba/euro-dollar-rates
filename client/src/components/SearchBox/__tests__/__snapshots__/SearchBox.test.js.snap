// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox rendered 1`] = `
<DocumentFragment>
  <div
    class="input-group input-group-lg"
  >
    <span
      class="input-group-text"
      id="inputGroup-sizing-lg"
    >
      Search Box
    </span>
    <input
      aria-describedby="inputGroup-sizing-lg"
      aria-label="Sizing example input"
      class="form-control"
      data-testid="search-input"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
